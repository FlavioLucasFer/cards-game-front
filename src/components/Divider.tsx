import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
    color: PropTypes.string,
    direction: PropTypes.oneOf(['horizontal', 'vertical']),
    margin: PropTypes.string,
};

export type DividerProps = PropTypes.InferProps<typeof propTypes>;

const Divider: React.FC<DividerProps> = props => {
    const { 
        color = '#585858', 
        direction = 'horizontal',
        margin,
    } = props; 
    
    if (direction === 'horizontal')
        return (
            <div 
                style={{
                    borderTop: `2px solid ${color}`,
                    margin: margin || '15px 0',
                }} 
            />
        );

    return (
        <div 
            style={{
                borderRight: `2px solid ${color}`,
                margin: margin || '0 15px',
            }} 
        />
    );
}

Divider.propTypes = propTypes;

export default Divider;
