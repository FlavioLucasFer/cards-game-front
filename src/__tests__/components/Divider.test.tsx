import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Divider from '../../components/Divider';

configure({ adapter: new Adapter() });

describe('Divider tests', () => {
    it('should render properly (horizontal)', () => {
        const wrapper = shallow(<Divider />);

        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should render properly (vertical)', () => {
        const wrapper = shallow(
            <Divider 
                direction='vertical'
            />
        );

        expect(wrapper.debug()).toMatchSnapshot();
    });
});
