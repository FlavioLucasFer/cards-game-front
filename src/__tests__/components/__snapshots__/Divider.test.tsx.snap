// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider tests should render properly (horizontal) 1`] = `"<div style={{...}} />"`;

exports[`Divider tests should render properly (vertical) 1`] = `"<div style={{...}} />"`;
